catkin_ws/src/kpr_testing/src/valves.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/python
+
+import rospy, termios, sys, contextlib, os
+import numpy as np
+from ur_msgs.srv import *
+from ur_msgs.msg import *
+
+header = """
+Check the status of I/O-pins
+\t-^D to exit
+\t-i to change income settings
+\t-p to send io states
+\t-t to toggle service succes
+"""
+
+def display():
+	os.system('clear')
+	print header
+#Making table
+	tableHeader = "Pin     |  State     Fun     "
+	print tableHeader
+	print "-"*len(tableHeader)
+	print "VacSens | ", info[VacSens][0], "  ", info[VacSens][1]
+	print "Gripper | ", info[Gripper][0], "  ", info[Gripper][1]
+
+#These IO pins are used
+VacSens = "DI1"
+Gripper = "DO2"
+
+#Set succes response
+response = True
+
+#dictionary of pins
+info = {VacSens:["-","     -"], Gripper:["-","     -"]}
```

Comment:
Declarations of variables and constants should be before any function declaration. Also please rename info to something like pinStates as info is ambiguous. The VacSens and Gripper variables are also unneeded when using `{"DI1" :["-","     -"], "DO2":["-","     -"]}`. To make the dictionary even more versatile, it could be altered to be of the form: `{"DI1" :("-","-","VacSens"), "DO2":("-","-","Gripper")}`. This way printing can be done even simpler: 
``` 
for entry in pinStates:
      e = pinStates[entry]
      print("{}({}) \t | {} \t {}".format(entry, e[2], e[0], e[1]))
```